server/planning/types/common.py

```diff
@@ -0,0 +1,267 @@
+    service: list[KeywordQCodeName] = Field(default_factory=list)
+    news_content_characteristics: list[KeywordNameValue] = Field(default_factory=list)
+    planning_ext_property: list[ExtProperty] = Field(default_factory=list)
+
+    # # Metadata hints.  See IPTC-G2-Implementation_Guide 16.5.1.1
+    by: list[str] = Field(default_factory=list)
+    credit_line: list[str] = Field(default_factory=list)
+    dateline: list[str] = Field(default_factory=list)
+
+    description_text: fields.HTML | None = None
+    genre: list[KeywordQCodeName] = Field(default_factory=list)
+    headline: fields.HTML | None = None
+
+    keyword: list[str] = Field(default_factory=list)
+    language: fields.Keyword | None = None
+    slugling: SlugLineField | None = None
+    subject: Annotated[
+        list[dict[str, Any]],
+        fields.elastic_mapping(
+            {
+                "type": "nested",
+                "include_in_parent": True,
+                "dynamic": False,
+                "properties": {
+                    "qcode": fields.Keyword,
+                    "name": fields.Keyword,
+                    "scheme": fields.Keyword,
+                },
+            }
+        ),
+    ] = Field(default_factory=list)
+
+    internal_note: fields.HTML | None = None
+    workflow_status_reason: str | None = None
+    priority: int | None = None
+
+
+@dataclass
+class NewsCoverageStatus:
+    # allows unknown
+    qcode: str | None = None
+    name: str | None = None
+    label: str | None = None
+
+
+@dataclass
+class CoverageAssignedTo:
+    assignment_id: fields.Keyword | None = None
+    state: fields.Keyword | None = None
+    contact: fields.Keyword | None = None
+
+    @classmethod
+    def to_elastic_properties(cls) -> dict[Literal["properties"], Any]:
+        """Generates the elastic mapping properties for the current dataclass"""
+
+        json_schema = TypeAdapter(cls).json_schema()
+        return json_schema_to_elastic_mapping(json_schema)
+
+
+@dataclass
+class CoverageFlags:
+    # allows unknown
+    no_content_linking: bool = False
+
+
+@dataclass
+class ScheduledUpdatePlanning:
+    internal_note: fields.HTML | None = None
+    contact_info: Annotated[fields.ObjectId | None, validate_data_relation_async("contacts")] = None
+    scheduled: datetime | None = None
+    genre: list[KeywordQCodeName] = Field(default_factory=list)
+    workflow_status_reason: str | None = None
+
+
+@dataclass
+class ScheduledUpdate:
+    scheduled_update_id: fields.Keyword | None = None
+    coverage_id: fields.Keyword | None = None
+    workflow_status: fields.Keyword | None = None
+    previous_status: fields.Keyword | None = None
+
+    assigned_to: CoverageAssignedTo | None = None
+    news_coverage_status: NewsCoverageStatus = Field(default_factory=dict)
+    planning: ScheduledUpdatePlanning = Field(default_factory=dict)
+
+
+@dataclass
+class PlanningCoverage:
+    # Identifiers
+    coverage_id: fields.Keyword
```

Comment:
Looks like this will also need to be optional, as we're generating one in the planning service it one is not defined

Reply:
done